File: akka-learning-java/src/test/java/org/spiral/java/sql/actor/AkkademyDbTest.java
package org.spiral.java.sql.actor;

import akka.actor.ActorRef;
import akka.actor.Props;
import akka.testkit.TestActorRef;
import org.junit.Test;
import org.spiral.java.sql.model.SetRequest;

import static org.junit.Assert.assertEquals;


/**
 * 消息处理单元测试
 *
 * @author dengguoqing
 * @date 2019/12/12
 * @copyright spiral
 * @since 1.0 Version
 */
public class AkkademyDbTest {

    @Test
    public void createReceive() {
        TestActorRef<AkkademyDb> actorRef = TestActorRef.create(Common.system,
                                                                Props.create(
                                                                        AkkademyDb.class));
        actorRef.tell(new SetRequest("key", "value"), ActorRef.noSender());

        AkkademyDb akkademyDb = actorRef.underlyingActor();
        assertEquals("value", akkademyDb.map.get("key"));
    }
}

File: akka-learning-scala/src/test/java/org/spiral/scala/sql/actor/AkkademyDbTest.scala
package org.spiral.scala.sql.actor

import akka.actor.ActorSystem
import akka.testkit.TestActorRef
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers
import org.spiral.scala.sql.model.SetRequest

/**
 * ${DESCRIPTION}
 *
 * @author dengguoqing
 * @date 2019/12/12
 * @since 1.0 Version
 * @copyright spiral
 */
class AkkademyDbTest extends AnyFlatSpec with Matchers {
  implicit val system = ActorSystem()


  it should "place key/value into map" in {
    val actorRef = TestActorRef(new AkkademyDb)
    actorRef ! SetRequest("key", "value")

    val akkademyDb = actorRef.underlyingActor
    akkademyDb.map.get("key").get should equal("value")
  }
}
